File: Data/StateMachine/Edge.cs
using System.Text.Json.Serialization;

namespace mcsim.Data.StateMachine
{
    public class Edge
    {
        public string Condition { get; }
        public Node Head { get; }
        public Node Tail { get; }
        public int Priority { get; }
        public string Actions { get; }

        [JsonConstructor]
        public Edge(string condition, Node head, Node tail, int priority, string actions)
        {
            this.Condition = condition;
            this.Head = head;
            this.Tail = tail;
            this.Priority = priority;
            this.Actions = actions;
        }
    }
}

File: mcsim-js/src/edge.mjs
export default class Edge {
    constructor(condition, head, tail, priority, actions) {
        this._condition = condition;
        this._head = head;
        this._tail = tail;
        this._priority = priority;
        this._actions = actions;
    }

    get condition() {
        return this._condition;
    }

    get head() {
        return this._head;
    }

    get tail() {
        return this._tail;
    }

    get priority() {
        return this._priority;
    }

    get actions() {
        return this._actions;
    }
}
